### PR TITLE
doc: mgmt/osdp: Fix doxygen comments

### PR DESCRIPTION
**Pull request by kartben**
- Misc fixes to improperly documented structs
- Show all PD and SC API in docs
- Document enums

### CODE REVIEW

File: include/zephyr/mgmt/osdp.h [hunk 1]
```diff
@@ -16,136 +21,189 @@
...
+	/**
+	 * Control code.
+	 * - 0 - no tone
+	 * - 1 - off
+	 * - 2 - default tone
+	 * - 3+ - TBD
+	 */
 	uint8_t control_code;
+	/**
+	 * The ON duration of the flash, in units of 100 ms.
```
[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/mgmt/osdp.h:144]: I don't know :D this wording is taken verbatim from the specification  :)

[the remark this answers is not in the document]
Comment (r2r0) [reply include/zephyr/mgmt/osdp.h:145]: Cargo cult ;-)
Example quote from spec: "The ON duration of the sound, in units of 100 ms. Must be nonzero unless the tone code is 0x01 (off.)"
So IMO "flash" word should be replaced by "sound" word.

[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/mgmt/osdp.h:145]: Damn, of course -- wrong struct!!

File: include/zephyr/mgmt/osdp.h [hunk 2]
```diff
@@ -16,136 +21,189 @@
...
- *             effect.
- * @param baud_rate baud rate value 9600/38400/115200
  */
 struct osdp_cmd_comset {
+	/**
+	 * Unit ID to which this PD will respond after the change takes effect.
+	 */
 	uint8_t address;
+	/**
+	 * Baud rate value 9600/38400/115200
```
[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/mgmt/osdp.h]: I am not sure what to do about this one -- I've changed to "Supported values: ..." but I am not sure that actually reflects the current implementation (and I don't think we're implementing OSDP 2.2 anyway?)

[the remark this answers is not in the document]
Comment (r2r0) [reply include/zephyr/mgmt/osdp.h]: Indeed - my fault.

[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/mgmt/osdp.h]: no worries. Please confirm that updated wording is correct / better than what was there before.

**Review by r2r0 — APPROVED** [review]

**Comment by kartben** [comment]
@sidcha could you please take a look? thanks!

**Review by sidcha — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**